.github/workflows/lint.yml

```diff
@@ -18,11 +18,11 @@ jobs:
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.x'
+          python-version: '3.11'
```

Comment:
@esoteric-ephemera May I ask why do we pin the python version for lint workflow? I believe just pin `mypy` version should suffice?

I would fix these new `mypy` errors in #4110, and I might revert this pin?

Reply:
Seemed to fix a lot of linting misfires but feel free to revert if I'm wrong. Not critical to this PR

Reply:
I guess the new linting issues is related to the new `mypy` 1.12 release (not the python version)

> Not critical to this PR

Absolutely, I would fix in #4110, thanks!